app/models/concerns/dossier_champs_concern.rb

```diff
@@ -274,6 +254,81 @@ def history
 
   private
 
+  def changed_champ_ids_for_merge(stream, overwrite_main: true)
+    buffer_champ_ids_h = champs.where(stream:, stable_id: revision_stable_ids)
+      .pluck(:id, :updated_at, :stable_id, :row_id)
+      .index_by { |(_, __, stable_id, row_id)| TypeDeChamp.public_id(stable_id, row_id) }
+      .transform_values { |(id, updated_at, _, __)| [id, updated_at] }
```

Comment:
proposition pour voir directement la forme du hash

```suggestion
      .pluck(:stable_id, :row_id, :id, :updated_at)
      .map { |(stable_id, row_id, id, updated_at)| [TypeDeChamp.public_id(stable_id, row_id), [id, updated_at] ] }
      .to_h
```